### PR TITLE
Check for both workflow and job names from Dr.CI

### PR DESCRIPTION
**Pull request by huydhn**
In https://github.com/pytorch/pytorch/pull/110362, the failure was flaky but merge bot treated it as an actual failure. This is a regression after https://github.com/pytorch/test-infra/pull/4604 where the name returned by Dr.CI now includes workflow name.  For example, the name is `trunk / macos-12-py3-arm64 / test (default, 2, 3, macos-m1-12)` in the JSON response:

```
{"FAILED": [], "FLAKY": [{"workflowId": 6372581477, "id": 17297638807, "name": "trunk / macos-12-py3-arm64 / test (default, 2, 3, macos-m1-12)", "jobName": "macos-12-py3-arm64 / test (default, 2, 3, macos-m1-12)", "conclusion": "failure", "completed_at": "2023-10-01T22:18:28Z", "html_url": "https://github.com/pytorch/pytorch/actions/runs/6372581477/job/17297638807", "head_branch": "ciflow/trunk/110362", "pr_number": 110362, "head_sha": "03f51e36dedf234931006d1db61677b229c9a119", "failure_captures": ["Failure: There is only 4671284KB free space left in /, which is less than the minimum requirement of"], "failure_line": "Failure: There is only 4671284KB free space left in /, which is less than the minimum requirement of 6291456KB for macOS", "time": "2023-10-01T22:17:53.847751Z"}], "BROKEN_TRUNK": [], "UNSTABLE": []}
```

I update merge bot to handle this better by considering both workflow name, job name, and the combination full name.

### CODE REVIEW

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -1588,9 +1588,17 @@ def is_flaky(
...
+    job_name = head_job.get("name", "")
+    full_name = f"{workflow_name} / {job_name}"
+
     # Consult the list of flaky failures from Dr.CI
     return any(
-        head_job["name"] == flaky["name"]
+        (
+            workflow_name == flaky["name"]
+            or job_name == flaky["name"]
+            or full_name == flaky["name"]
```
Comment (clee2000) [inline .github/scripts/trymerge.py:1600]: why is it not just always full_name?

Comment (huydhn) [reply .github/scripts/trymerge.py:1600]: Does `full_name` alway work with those like "Check Label", "BC Lint", and some workflow with only workflow name IIRC?  So I check for all three like a safer option?

Comment (huydhn) [reply .github/scripts/trymerge.py:1600]: If the scheme `workflow name / job name` applies to them all, then full_name is indeed enough.  Figuring out the correct name here gives me a headache, so I just check for them all to be on the safe side lol

Comment (huydhn) [reply .github/scripts/trymerge.py]: Confirm that the full name is sufficient.  Every job follows the same pattern, i.e.

```
Labeler || triage
Check Labels || Check labels
BC Lint || bc_linter
Auto Request Review || Auto Request Review
```

**Review by clee2000 — APPROVED** [review]

**Pull request closed without merging**